### PR TITLE
chore/OcTable add draggable functionality to table rows

### PR DESCRIPTION
**Pull request by otaviolbarbosa**
## Summary by CodeRabbit

- **New Features**
    - Enhanced the table component with improved customization options for the table header and body sections.

### CODE REVIEW

**Review by Sammuel09 — APPROVED** [review]
LGTM

File: packages/@orchidui-vue/src/DataDisplay/Table/OcTable.vue [hunk 1]
```diff
@@ -30,14 +31,21 @@ const props = defineProps({
     type: Boolean,
     default: false,
   },
+  isDraggable: {
+    type: Boolean,
+    default: false,
+  },
```
[the remark this answers is not in the document]
Comment (Sammuel09) [reply packages/@orchidui-vue/src/DataDisplay/Table/OcTable.vue:37]: @otaviolbarbosa please review

**Review by HoaiLee — CHANGES_REQUESTED** [review]
Please use `slot` instead
- Draggable should not be apart of Table component

File: packages/@orchidui-vue/src/DataDisplay/Table/OcTable.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (otaviolbarbosa) [reply packages/@orchidui-vue/src/DataDisplay/Table/OcTable.vue:37]: Default should be false

File: packages/@orchidui-vue/src/DataDisplay/Table/OcTable.vue [hunk 2]
```diff
@@ -199,100 +215,120 @@ onMounted(() => onScroll());
...
-        v-for="(field, i) in fields"
-        :key="i"
-        :class="{
-          'border-b': fields.length !== i + 1,
-        }"
+      <Draggable
+        v-slot="{ list }"
+        :model-value="fields"
+        class="grid gap-3"
+        @update:model-value="$emit('update:modelValue', $event)"
```
Comment (HoaiLee) [inline packages/@orchidui-vue/src/DataDisplay/Table/OcTable.vue:222]: here we need to wrap the content inside slot with name 
```suggestion
      <slot
        name="table-body"
        :fields="fields"
        :is-scroll-on-end="isScrollOnEnd"
        :on-click-row="onClickRow"
        :another-props="anotherPropsCanPassIntoParentComponents"
      >
        <div
          v-for="(field, i) in fields"
          :key="i"
          :class="{
            'border-b': fields.length !== i + 1
```

Comment (HoaiLee) [reply packages/@orchidui-vue/src/DataDisplay/Table/OcTable.vue:222]: In parent component, which we need to use with slot
```
<oc-table
  :options="options"
  :is-sticky="true"
>
  <template #table-body="{ fields, onClickRow , isScrollOnEnd, anotherProps, ...etc }">

    // Here we do custom slot
    <Draggable
        v-slot="{ list }"
        :model-value="fields"
        class="grid gap-3"
        @update:model-value="doSomething"
    >
      <div
        v-for="(field, i) in fields"
        :key="i"
      >
        <TableCell
          v-for="(header, j) in headers"
          :key="`${j}-${i}`"
        >
          // cell data here
        </TableCell>
      </div>
    </Draggable>
  </template>
</oc-table
```

**Review by HoaiLee — CHANGES_REQUESTED** [review]

File: packages/@orchidui-vue/src/DataDisplay/Table/OcTable.vue (hunk 2 above)
Comment (otaviolbarbosa) [reply packages/@orchidui-vue/src/DataDisplay/Table/OcTable.vue]: @HoaiLee @kewcoder added table-header and table-body slots. Please, take a look.

**Review by HoaiLee — APPROVED** [review]

**Comment by HoaiLee** [comment]
Good to have `slot` 👍

**Pull request merged**